scriptworker/test/test_worker.py

```diff
@@ -60,27 +61,32 @@ def test_main(mocker, context, event_loop):
 def test_main_sigterm(mocker, context, event_loop):
     """Test that sending SIGTERM causes the main loop to stop after the next
     call to async_main."""
-    config = dict(context.config)
-    config['poll_interval'] = 1
-    creds = {'fake_creds': True}
-    config['credentials'] = deepcopy(creds)
+    run_tasks_cancelled = event_loop.create_future()
 
-    async def async_main(*args):
+    class MockRunTasks:
```

Comment:
There's a `constants helpers and fixtures` section at the top; do these belong there?

Reply:
I'm going to leave this one here: It's only used in the one test, and has `cancel()`'s behaviour set accordingly. I'm not sure it's useful to share, and I think there's benefit in it being within the test definition: it hints to the next dev that it's only used in that one test.

Let me know what you think